### PR TITLE
Adding mkdocs

### PR DESCRIPTION
**Pull request by daviddavis**
Here's an option to build and publish our docs. Feedback welcome. If this looks good, I can work on a github action workflow to publish our docs to docs.pulpproject.org. 

Also, one thing I need to figure out is how to get the links working.

### CODE REVIEW

**Comment by fao89** [comment]
we use mkdocs on pulp-installer and pulp-operator, which links are not working?

**Comment by daviddavis** [comment]
I'm currently symlinking the README.md at `docs/index.md` and all the relative links in README.md to places like `docs/install.md` don't work.

I think I should maybe break up README.md into files like `contributing.md` and create an actual `docs/index.md` instead of symlinking the README at `docs/index.md`.

**Comment by fao89** [comment]
> I'm currently symlinking the README.md at `docs/index.md` and all the relative links in README.md to places like `docs/install.md` don't work.
> 
> I think I should maybe break up README.md into files like `contributing.md` and create an actual `docs/index.md` instead of symlinking the README at `docs/index.md`.

we did something similar on the installer on the installer, all roles are symlinked, here is an example of pulp_all_services Readme.md referring to pulp_services:
```
- [pulp_services](../pulp_services/): A role to install & configure Pulp 3's
```

**Comment by daviddavis** [comment]
@pulp/cli I fixed the links. Please test this out and let me know what you think.

To look at the docs, check out this PR and run `make servedocs`.

File: README.md [hunk 1]
```diff
@@ -4,117 +4,6 @@ This is a command line interface for Pulp 3.
...
-```
-
-Read [the installation and configuration doc](docs/install.md) for more information.
-
-## Known issues
-
-  * Redirecting from `http` to `https`, as done by a typical Pulp installation,
-    does not work properly with `POST` and `PUT` requests.
-    Please use `https://` in the base url.
-    Note that the attempt to use `http` leaks sensitive data over an unencrypted connection.
```
Comment (daviddavis) [inline README.md:36]: FYI, I dropped this. I don't think it's an issue anymore since we don't follow redirects.

Comment (mdellweg) [reply README.md:36]: So what happens now, if you try to use the wrong schema?

Comment (daviddavis) [reply README.md:36]: You get a connection error.

Comment (daviddavis) [reply README.md:36]: ^ That's if there's no redirect.

If there's a redirect, then users receive [this error](https://github.com/pulp/pulp-cli/blob/42f62321e61465b28c9dcbc0c9822538631dcb16/pulpcore/cli/common/openapi.py#L290-L294)

**Comment by daviddavis** [comment]
@pulp/cli it looks like both pulp_installer and pulp-operator are using mkdocs as well but they just publish to https://readthedocs.org/. I could set that up and it would be a lot less work than using docs.pulpproject.org. However, I feel like the latter is a better location. Thoughts?

**Comment by ggainey** [comment]
> @pulp/cli it looks like both pulp_installer and pulp-operator are using mkdocs as well but they just publish to https://readthedocs.org/. I could set that up and it would be a lot less work than using docs.pulpproject.org. However, I feel like the latter is a better location. Thoughts?

We're trying to deprecate/remove all readthedocs and consolidate on docs.pulpproject, yeah? I'd like everything to be in one place.

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Pulp 3 CLI Docs
+
+This is the documentation for Pulp 3's command line interfact.
```
Comment (mdellweg) [inline docs/index.md:3]:
```suggestion
This is the documentation for Pulp 3's command line interface.
```

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Releasing (for internal use)
+
+1. Generate the changelog (eg `towncrier --yes --version 0.1.0`) and commit
+1. Run `bumpversion release`, commit your local changes, and note the commit sha
+1. Run `bumpversion minor` to update the version to the next dev release version
+1. Push your commits, open a PR, and get it merged
+1. After your PR is merged, pull the latest changes from develop
+1. Now tag your release commit (e.g. `git tag 0.1.0`) and push to pulp/pulp-cli
+1. Monitor the build job and then check PyPI to make sure the package has been uploaded
```
Comment (mdellweg) [inline docs/releasing.md:9]: Those jobs will publish the docs, right?
```suggestion
1. Monitor the build job and then check PyPI to make sure the package has been uploaded and the docs updated
```

**Review by mdellweg — COMMENTED** [review]
A typo and a suggestion.
Great piece of work!

**Review by mdellweg — APPROVED** [review]
I think, we should merge this soon, to make the "edit on GH" banner work.
Hopefully it will lead to further improvements of the docs.

**Pull request merged**